sklearn/tests/test_isotonic.py

```diff
@@ -535,3 +535,15 @@ def test_isotonic_thresholds(increasing):
         assert all(np.diff(y_thresholds) >= 0)
     else:
         assert all(np.diff(y_thresholds) <= 0)
+
+
+def test_isotonic_strict():
+    # check on enforcing strict monotonicity
+    X = np.array([0.1, 0.3, 0.5, 0.7, 1.])
+    y = np.array([1, 0, 1, 0, 1])
+
+    ireg = IsotonicRegression(strict=True)
+    ireg.fit(X, y)
+    pred = ireg.predict([0.1, 0.2, 0.3])
+
+    assert all(np.diff(pred) != 0)
```

Comment:
Let's check `> 0`
also I'd suggest checking with `increasing=False`

Reply:
We should also check  `X_thresholds_` and  `y_thresholds_` which are affected IIUC

Reply:
Done, I've also added a test on `increasing=False` that actually raises a `ValueError` when it's impossible to ensure strict monotonicity.